docs/developers/deprecating.md

```diff
@@ -0,0 +1,45 @@
+#
+
+## What is deprecation?
+
+Whenever a feature of iRODS needs to be removed for whatever reason, we follow a process which first deprecates the feature and then removes it. Here are some possible reasons for deprecating a feature...
+
+ - It has some issue which can only be fixed by replacement.
+ - It has been or is being replaced by a better alternative.
+ - Its purpose is unknown or no longer supported.
+
+In general, users and administrators should expect the following process when a server feature is being deprecated:
+
+ 1. A feature can be declared as deprecated in any minor version.
+ 2. The feature will continue to be supported for all major versions in which it is deprecated (and previous major versions in which the feature exists).
+ 3. The feature can be removed in the next major version, but will not necessarily be removed. It may remain in a deprecated state in the next major version.
+ 4. If a feature is shown to be worth keeping after it has been deprecated and before it has been removed, it can be de-deprecated in any minor version.
+
+These also apply to releases of the iCommands package because they are strictly tied to releases of the server.
+
+## How do I deprecate a feature?
+
+We will now describe the steps which should be taken to deprecate various types of features. In general, we seek to avoid changing or removing runtime functionality.
+
+The following things need to be done whenever a feature is being declared as deprecated:
+
+ 1. Update documentation to include alternative, supported way(s) of accomplishing what is now deprecated. If no alternative is available, include a note indicating that use of the deprecated API should be avoided. Updating the documentation includes adding a `\deprecated` tag with the aforementioned notes to the Doxygen for any public declarations associated with the feature.
+ 2. Add a "deprecated" attribute specifier (see [https://en.cppreference.com/w/cpp/language/attributes/deprecated](https://en.cppreference.com/w/cpp/language/attributes/deprecated)) to the C/C++ public declarations specific to this API and include a message that is similar if not identical to the message used in (1). This includes any functions, structs, classes, and typedefs. If a struct, class, or typedef is being used by other APIs which are not being deprecated and the usage is not related to the feature which is being deprecated, do not deprecate these. For example, `DataObjInp` could be used by an API which is being deprecated, but `DataObjInp` is not specific to that API. Therefore, `DataObjInp` would not be a candidate for deprecation.
```

Comment:
Is there a link for the `__attribute__` syntax for C APIs?

Reply:
I tried to find something, but all I could find was the C23 equivalent which is also just `[[deprecated]]`: https://en.cppreference.com/w/c/language/attributes/deprecated Should I include that?

Reply:
Let's instruct the reader to use the `__attribute__` syntax for C code. No links necessary. They can google how to do it. If it's incorrect, we'll catch it during code review.

Reply:
I just removed the link and mentioned what should be used. I'll resolve this but feel free to open a new conversation to continue tweaking, or... just comment here. Either way.